contracts/Pod.sol

```diff
@@ -472,59 +481,64 @@ contract Pod is Initializable, ERC20Upgradeable, OwnableUpgradeable, IPod {
     /**
      * @dev The internal function for the public withdraw function, which calculates a user's token allocation from burned shares.
      * @param shares Amount of "token" deposited into the Pod.
+     * @param maxFee Max fee amount for withdrawl if amount isn't available in float.
      * @return uint256 The token amount returned for the burned shares.
      */
-    function _burnShares(uint256 shares) internal returns (uint256) {
+    function _burnSharesAndGetTokensReturned(uint256 shares, uint256 maxFee)
+        internal
+        returns (uint256)
+    {
         // Calculate Percentage Returned from Burned Shares
         uint256 amount = (balance().mul(shares)).div(totalSupply());
 
         // Burn Shares
         _burn(msg.sender, shares);
 
         // Check balance
-        IERC20Upgradeable _token = IERC20Upgradeable(token);
-        uint256 currentBalance = _token.balanceOf(address(this));
+        uint256 currentBalance = token.balanceOf(address(this));
 
         // Withdrawl Exceeds Current Token Balance
         if (amount > currentBalance) {
             // Calculate Withdrawl Amount
-            uint256 _withdraw = amount.sub(currentBalance);
+            uint256 withdraw = amount.sub(currentBalance);
 
             // Withdraw from Prize Pool
-            uint256 exitFee = _withdrawFromPool(_withdraw);
+            uint256 exitFee = _withdrawFromPool(withdraw, maxFee);
 
             // Add Exit Fee to Withdrawl Amount
             amount = amount.sub(exitFee);
         }
 
-        // Transfer Deposit Token to Message Sender
-        _token.safeTransfer(msg.sender, amount);
-
         // Return Token Withdrawl Amount
         return amount;
     }
 
     /**
      * @dev Withdraws from Pod prizePool if the float balance can cover the total withdraw amount.
-     * @param _amount Amount of tokens to withdraw in exchange for the tickets transfered.
+     * @param amount Amount of tokens to withdraw in exchange for the tickets transfered.
+     * @param maxFee Max fee amount for withdrawl if amount isn't available in float.
      * @return uint256 The exit fee paid for withdraw from the prizePool instant withdraw method.
      */
-    function _withdrawFromPool(uint256 _amount) internal returns (uint256) {
-        IPrizePool _pool = IPrizePool(_prizePool);
-
+    function _withdrawFromPool(uint256 amount, uint256 maxFee)
+        internal
+        returns (uint256)
+    {
         // Calculate Early Exit Fee
         (uint256 exitFee, ) =
```

Comment:
This check isn't necessary, as the prize pool will revert if the passed max exit fee is less than the actual exit fee.